tests/internal/processor_conditional.c

```diff
@@ -51,21 +51,35 @@ static void cleanup_test_resources(struct flb_config *config, struct flb_process
             flb_sds_destroy(pu->name);
         }
         pthread_mutex_destroy(&pu->lock);
-        
+
         if (pu->parent) {
             /* Remove from parent's list */
             mk_list_del(&pu->_head);
             flb_processor_destroy(pu->parent);
         }
-        
+
         flb_free(pu);
     }
-    
+
     if (config) {
         flb_config_exit(config);
     }
 }
 
+/* Create a string variant for testing non-kvlist condition handling */
+static struct cfl_variant *create_string_condition_variant(const char *value)
+{
+    struct cfl_variant *variant;
+
+    /* Using cfl_variant_create_from_string instead of a non-existent cfl_variant_set_string */
+    variant = cfl_variant_create_from_string((char *)value);
+    if (!variant) {
```

Comment:
If you are not doing anything (ie. reporting) if `cfl_variant_create_from_string` returns `NULL` why do you have this code? returning `variant` would be the same.